demo/site/src/blocks/YouTubeVideoBlock.tsx

```diff
@@ -13,19 +13,23 @@ const getHeightInPercentForAspectRatio = (aspectRatio: YouTubeVideoBlockData["as
     }
 };
 
+const EXPECTED_YT_ID_LENGTH = 11;
+
+const parseYoutubeIdentifier = (value: string): string | undefined => {
+    // regex from https://stackoverflow.com/a/51870158
+    const regExp =
+        /(https?:\/\/)?(((m|www)\.)?(youtube(-nocookie)?|youtube.googleapis)\.com.*(v\/|v=|vi=|vi\/|e\/|embed\/|user\/.*\/u\/\d+\/)|youtu\.be\/)([_0-9a-zA-Z-]+)/;
+    const match = value.match(regExp);
+    const youtubeId = value.length === EXPECTED_YT_ID_LENGTH ? value : match && match[8].length == EXPECTED_YT_ID_LENGTH ? match[8] : null;
+
+    return youtubeId ?? undefined;
+};
+
 const YouTubeVideoBlock: React.FunctionComponent<PropsWithData<YouTubeVideoBlockData>> = ({
     data: { youtubeIdentifier, autoplay, loop, showControls, aspectRatio },
 }) => {
-    try {
-        const url = new URL(youtubeIdentifier);
-        const searchParams = url.searchParams;
-        if (!searchParams.has("v")) {
-            throw new Error("URL has no ID (v) param");
-        }
-        youtubeIdentifier = searchParams.get("v") as string;
-    } catch (error) {
-        // no url, but ID was specified
-    }
+    const identifier = parseYoutubeIdentifier(youtubeIdentifier);
+    youtubeIdentifier = identifier ?? "";
```

Comment:
```suggestion
    youtubeIdentifier = parseYoutubeIdentifier(youtubeIdentifier) ?? "";
```

Should we really provide "" as default value here?

Reply:
We need it for the url but i refactored it so it doesn't seem like a default value